### PR TITLE
Add support for validating specified files

### LINKED ISSUE #30: Any way to specify a list of files to validate?
I am in need of this in order to validate only the files that have changed in the PR instead of validating the whole repo. Is there any way that this can be added?

### PR DESCRIPTION
**Pull request by thecodrr**
Closes #30

### CODE REVIEW

**Comment by thecodrr** [comment]
@GrantBirki sorry for the ping but would love to get this merged. A PR I am working on is stuck because of this: https://github.com/streetwriters/notesnook-themes/pull/7

**Comment by GrantBirki** [comment]
Another excellent PR, thank you! And the `@` ping is actually a requirement for me to see anything because my GitHub notifications are a disaster so please `@` me everytime 😂

**Comment by thecodrr** [comment]
@GrantBirki No problem! This is a really useful project and I love how clean and simple it is to extend & add new things on top of.

**Comment by GrantBirki** [comment]
@thecodrr Could you please do the following:

- Run `npm run all` and then commit your changes (need to bundle as the last commit for Actions packages)
- Update the [`action.yml`](https://github.com/GrantBirki/json-yaml-validate/blob/main/action.yml) file to reflect changes for this new `files` input option
- Also update the main [inputs](https://github.com/GrantBirki/json-yaml-validate#inputs-) table to reflect these changes

**Comment by GrantBirki** [comment]
From your issue:

> I am in need of this in order to validate only the files that have changed in the PR instead of validating the whole repo. Is there any way that this can be added?

You will certainly find the [GrantBirki/git-diff-action](https://github.com/GrantBirki/git-diff-action) extremely useful as it can be used to parse a Git diff (files changed in a PR) into machine readable JSON and it is a GitHub Action. I used this Action **all the time** to validate changes to files that were made only in a PR and not _every where else_. You should totally give it a look 👀

**Comment by thecodrr** [comment]
@GrantBirki done

File: src/functions/json-validator.js [hunk 1]
```diff
@@ -45,6 +45,7 @@ export async function jsonValidator(exclude) {
   const yamlExtension = core.getInput('yaml_extension')
   const yamlExtensionShort = core.getInput('yaml_extension_short')
   const useDotMatch = core.getBooleanInput('use_dot_match')
+  let files = core.getInput('files').split('\n').filter(Boolean)
```
Comment (GrantBirki) [inline src/functions/json-validator.js:48]:
```suggestion
  let files = core.getMultilineInput('files')
```

I think you could leverage `core.getMultilineInput()` here.

Example source in the `actions/checkout` project: https://github.com/actions/checkout/blob/96f53100ba2a5449eb71d2e6604bbcd94b9449b5/src/input-helper.ts#L86

Then you could do something like this for end users:

Screenshot 2023-08-11 at 12 13 01 PM

---

[source code](https://github.com/actions/toolkit/blob/45c49b09df04cff84c5f336f07d5232fa7103761/packages/core/src/core.ts#L140-L161)

Comment (GrantBirki) [reply src/functions/json-validator.js:48]: I have not used the `core.getMultilineInput()` before but I think that should help you out here

Comment (thecodrr) [reply src/functions/json-validator.js]: Done

File: README.md [hunk 2]
```diff
@@ -37,6 +37,7 @@ Here is a quick example of how to install this action in any workflow:
 | `mode` | `false` | `"fail"` | The mode to run the action in `"warn"` or `"fail"` |
 | `comment` | `false` | `"false"` | Whether or not to comment on a PR with the validation results - `"true"` or `"false"` |
 | `base_dir` | `false` | `"."` | The base directory to search for JSON and YAML files (e.g. ./src) - Default is `"."` which searches the entire repository |
+| `files` | `false` | `""` | List of file paths to validate. Each file path must be on a newline. |
```
Comment (GrantBirki) [inline README.md:40]: Would both of these examples work?

**Example 1**:

```yaml
- name: json-yaml-validate
  id: json-yaml-validate
  uses: GrantBirki/json-yaml-validate@vX.X.X
  with:
    files: "test1.json\ntest2.json\ntest3.json"
```

**Example 2**:

```yaml
- name: json-yaml-validate
  id: json-yaml-validate
  uses: GrantBirki/json-yaml-validate@vX.X.X
  with:
    files: |
      test1.json
      test2.json
      test3.json
```

Comment (GrantBirki) [reply README.md:40]: Mainly just curious, thanks!

Comment (thecodrr) [reply README.md:40]: I don't think the first will work. What's the best way to check?

Comment (GrantBirki) [reply README.md:40]: I'll merge and we can give it a go! If only the second option works, that totally fine. More curiosity than anything else.

**Review by GrantBirki — APPROVED** [review]

**Comment by Aficionado45** [comment]
@thecodrr How do you propose to pass the input as file paths in a new line dynamically reading from files changed in a PR. Since we can't set up a GITHUB_ENV or GITHUB_OUTPUT of a string with values separated by \n. Can you share any example code.

**Comment by Aficionado45** [comment]
@GrantBirki Can you please address above comment

**Pull request merged**